sync/src/sync_io.rs

```diff
@@ -66,19 +70,22 @@ pub struct NetSyncIo<'s, 'h> where 'h: 's {
 	chain: &'s BlockChainClient,
 	snapshot_service: &'s SnapshotService,
 	chain_overlay: &'s RwLock<HashMap<BlockNumber, Bytes>>,
+	private_tx_provider: Option<Arc<PrivateTransactionProvider>>,
 }
 
 impl<'s, 'h> NetSyncIo<'s, 'h> {
 	/// Creates a new instance from the `NetworkContext` and the blockchain client reference.
 	pub fn new(network: &'s NetworkContext<'h>,
 		chain: &'s BlockChainClient,
 		snapshot_service: &'s SnapshotService,
+		private_tx_provider: Option<Arc<PrivateTransactionProvider>>,
 		chain_overlay: &'s RwLock<HashMap<BlockNumber, Bytes>>) -> NetSyncIo<'s, 'h> {
 		NetSyncIo {
 			network: network,
 			chain: chain,
 			snapshot_service: snapshot_service,
 			chain_overlay: chain_overlay,
+			private_tx_provider: private_tx_provider,
```

Comment:
why is it optional? It's always initialized to some, even in tests

Reply:
No, you're not correct. See chain.rs tests. All of them don't have Private Provider and initialize it with None.